bait-files/src/main/java/com/google/sps/servlets/RequestsServlet.java

```diff
@@ -32,6 +32,10 @@ public int compare(Request a, Request b) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    boolean illegalAccess = Boolean.parseBoolean(request.getParameter("username"));
+    if (illegalAccess) {
+        throw new IllegalAccessError("The resource you are trying to access is restricted");
+    }
```

Comment:
I'm unclear on what is happening here... if the request contains a parameter called "username", then this is illegal access?  Why?
Trying to guess here... but if the code is saying that any request with "username" populated must have come from a normal user, but this code only works for admit user, I'd rename the boolean variable to isNonAdminUser.

Reply:
But also, if that is the case, we don't want to trust the parameters in the request to tell us whether or not the user has privileges.  Anyone with an internet connection can theoretically write their own request and send it to your server and add the parameter iAmCool=true (or whatever equivalent you might be using).

I think your filter should be taking care of illegal access, right?  I would say check and handle the user permissions there and then don't even check anything here, since that's what the filter is for.  Otherwise you'd be checking user access at the beginning of every doGet and doPost everywhere.

Reply:
Remove

Reply:
Ah, I have a new guess about what's going on... did you mean to say request.getParameter("illegalAccess")?  And you manually set it to true in the filter if the user doesn't have access?  That could work, I think.  I'd still rename the boolean variable to something with a verb in it.  Like isIllegalAccess.  Or userDoesNotHavePermission.